lib/spack/spack/test/config.py

```diff
@@ -277,6 +277,28 @@ def test_add_config_path(mutable_config):
     compilers = spack.config.get("packages")["all"]["compiler"]
     assert "gcc" in compilers
 
+    # Try quotes to escape brackets
+    path = "config:install_tree:projections:cmake:\
+'morepadding/{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'"
+    spack.config.add(path)
+    set_value = spack.config.get("config")["install_tree"]["projections"]["cmake"]
+    assert (
+        set_value
+        == "morepadding/{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}"
+    )
```

Comment:
Apologies one additional thing: as it is, this test will actually succeed regardless of whether our regex capture gathers quotes, because yaml will not be confused if the first character is not `{` (or `[`, i.e. something of significance to YAML). To make this test significant, it should remove the "morepadding/" part and just try to set the string `"{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}"`.